code/modules/organs/blood.dm

```diff
@@ -299,22 +299,30 @@ proc/blood_splatter(var/target,var/datum/reagent/organic/blood/source,var/large)
 	return 100
 
 /mob/living/carbon/human/get_blood_circulation()
-	var/obj/item/organ/internal/heart/heart = internal_organs_by_name[BP_HEART]
+	var/heart_efficiency = get_organ_efficiency(OP_HEART)
+	var/robo_check = TRUE	//check if all hearts are robotic
+	var/open_check = FALSE  //check if any heart is open
+	for(var/obj/item/organ/internal/heart/heart in organ_list_by_process(OP_HEART))
+		if(!(BP_IS_ROBOTIC(heart)))
+			robo_check = FALSE
+		if(heart.open)
+			open_check = TRUE
+
 	var/blood_volume = get_blood_volume()
-	if(!heart || (heart.pulse == PULSE_NONE && !(status_flags & FAKEDEATH) && !BP_IS_ROBOTIC(heart)))
+	if(!heart_efficiency || (pulse == PULSE_NONE && !(status_flags & FAKEDEATH) && !robo_check))
```

Comment:
I did a review similar to this one. instead of "!heart" you could do: heart_efficiency  <= 0.

Reply:
Fixed.